internal/controller/resources/resources.go

```diff
@@ -0,0 +1,104 @@
+package resources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/dana-team/nfspvc-operator/internal/controller/utils"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	danaiov1alpha1 "github.com/dana-team/nfspvc-operator/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var FailedCleanupError = errors.New("failed nfspvc cleanup")
+
+// HandleDelete ensures the deletion of the nfspvc.
+func HandleDelete(ctx context.Context, nfspvc danaiov1alpha1.NfsPvc, k8sClient client.Client) (bool, error) {
+	if controllerutil.ContainsFinalizer(&nfspvc, utils.NfsPvcDeletionFinalizer) {
+		pvcDeleted, pvDeleted, err := areResourceDeleted(ctx, nfspvc, k8sClient)
+		if err != nil {
+			return false, err
+		}
+		if pvDeleted && pvcDeleted {
+			return true, nil
+		} else if !pvDeleted && pvcDeleted {
+			if err := cleanup(ctx, nfspvc.Name, nfspvc.Namespace, k8sClient); err != nil {
+				return false, err
+			}
+			pvName := nfspvc.Name + "-" + nfspvc.Namespace + "-pv"
+			return false, fmt.Errorf("the pv %s is not deleted yet, %w", pvName, FailedCleanupError)
```

Comment:
For names, use `%q` instead of `%s` since it makes it easier to read. Also use `:` instead of `,`:

```go
fmt.Errorf("pv %q has not been deleted yet: %w", pvName, FailedCleanupError)
```